test/integration/Resources/PaypageV2Test.php

```diff
@@ -0,0 +1,309 @@
+
+        $this->getUnzerObject()->createPaypage($paypage);
+
+        $this->assertCreatedPaypage($paypage);
+    }
+
+    /**
+     * @test
+     * @group skip
+     */
+    public function createPaypageWithUrls()
+    {
+        $urls = new Urls();
+        $urls->setTermsAndCondition('https://termsandcondition.com');
+        $urls->setPrivacyPolicy('https://privacypolicy.com');
+        $urls->setImprint('https://imprint.com');
+        $urls->setHelp('https://help.com');
+        $urls->setContact('https://contact.com');
+        $urls->setFavicon('https://favicon.com');
+        $urls->setReturnSuccess('https://returnsuccess.com');
+        $urls->setReturnPending('https://returnpending.com');
+        $urls->setReturnFailure('https://returnfailure.com');
+        $urls->setReturnCancel('https://returncancel.com');
+
+        $paypage = new Paypage(9.99, 'EUR', 'charge');
+        $paypage->setUrls($urls);
+
+        $this->getUnzerObject()->createPaypage($paypage);
+
+        $this->assertCreatedPaypage($paypage);
+    }
+
+    /**
+     * @test
+     */
+    public function createPaypageWithStyle()
+    {
+        $style = new Style();
+        $style->setFontFamily('comic sans');
+        $style->setButtonColor('red');
+        $style->setPrimaryTextColor('green');
+        $style->setLinkColor('blue');
+        $style->setBackgroundColor('black');
+        $style->setCornerRadius('5px');
+        $style->setShadows(true);
+        $style->setHideUnzerLogo(true);
+
+        $paypage = new Paypage(9.99, 'EUR', 'charge');
+        $paypage->setStyle($style);
+
+        $this->getUnzerObject()->createPaypage($paypage);
+        $this->assertCreatedPaypage($paypage);
+    }
+
+    /** @test
+     * @dataProvider paymentMethodsConfigsDataProvider
+     **/
+    public function createPaypageWithMethodConfigs(PaymentMethodsConfigs $configs)
+    {
+        $paypage = new Paypage(9.99, 'EUR', 'charge');
+        $paypage->setPaymentMethodsConfigs($configs);
+        $this->getUnzerObject()->createPaypage($paypage);
+
+        $this->assertCreatedPaypage($paypage);
+    }
+
+    /** @test * */
+    public function createPaypageWithMethodResourceIds()
+    {
+        $unzer = $this->getUnzerObject();
+        $customer = $unzer->createCustomer($this->getMinimalCustomer());
+        $basket = $this->createV2Basket();
+        $metadata = $unzer->createMetadata((new Metadata())->setShopType('unitTests'));
+
+
+        $resources = new Resources($customer->getId(), $basket->getId(), $metadata->getId());
+        $paypage = new Paypage(9.99, 'EUR', 'charge');
+        $paypage->setResources($resources);
+        $unzer->createPaypage($paypage);
+
+        $this->assertCreatedPaypage($paypage);
+    }
+
+    /** @test
+     * @group skip
+     */
+    public function createPaypageWithRiskData()
+    {
+        $unzer = $this->getUnzerObject();
+        $risk = new RiskData();
+
+        $risk->setThreatMetrixId('f544if49wo4f74ef1x')
```

Comment:
We only support these fields in risk:
 "risk": {
    "registrationLevel": "string",
    "registrationDate": "string",
    "customerGroup": "top",
    "confirmedOrders": 0,
    "confirmedAmount": 0
  }

customerId and threatmetrixId are gone.
ThreadMetrixId will be generated by the Paypage UI

Reply:
@AntonAtUnzer
I see, I reused the risk object for normal transactions here. So i can not remove them from the original class.
The merchant could just not set the properties. Or should i make sure that he is not setting them for paypage requests?